tests/Unit/Domain.LinnApps.Tests/PurchaseOrderDeliveryServiceTests/WhenUpdatingDetailsAddingToDeliveryTotal.cs

```diff
@@ -0,0 +1,163 @@
+                                                     AvailableAtSupplier = "N"
+                                                 },
+                                                 new PurchaseOrderDelivery
+                                                 {
+                                                     DeliverySeq = 1,
+                                                     OurDeliveryQty = 10,
+                                                     BaseOurUnitPrice = 11,
+                                                     DateRequested = 25.March(2022),
+                                                     AvailableAtSupplier = "N"
+                                                 },
+
+                                         };
+
+
+            // Order will be updated to delivery total. Simulate adding deliveries.
+            this.line = new PurchaseOrderDetail
+            {
+                Line = 1,
+                OrderUnitPriceCurrency = 666,
+                OrderConversionFactor = 1.5m,
+                BaseOurUnitPrice = 111,
+                PurchaseDeliveries = new List<PurchaseOrderDelivery>(),
+                OurQty = 10,
+                OrderQty = 10
+            };
+            this.order = new PurchaseOrder
+            {
+                OrderNumber = 123456,
+                Details = new List<PurchaseOrderDetail> { this.line },
+                SupplierId = 9876543,
+                DocumentTypeName = "PO"
+            };
+            this.PurchaseOrderRepository.FindById(this.order.OrderNumber).Returns(this.order);
+
+            this.PurchaseOrdersPack.GetVatAmountSupplier(
+                this.line.BaseOurUnitPrice.GetValueOrDefault() * this.updatedDeliveries.First().OurDeliveryQty.GetValueOrDefault(),
+                this.order.SupplierId).Returns(this.baseVatAmount);
+
+            this.PurchaseOrdersPack.GetVatAmountSupplier(
+                this.line.OrderUnitPriceCurrency.GetValueOrDefault()
+                * this.updatedDeliveries.First().OurDeliveryQty.GetValueOrDefault(),
+                this.order.SupplierId).Returns(this.vatAmount);
+
+            this.AuthService.HasPermissionFor(
+                    AuthorisedAction.PurchaseOrderUpdate, Arg.Any<IEnumerable<string>>())
+                .Returns(true);
+
+            this.MiniOrderRepository.FindById(this.order.OrderNumber).Returns(new MiniOrder { });
+            this.result = this.Sut.UpdateDeliveriesForOrderLine(
+                this.order.OrderNumber,
+                this.line.Line,
+                this.updatedDeliveries,
+                new List<string>());
+        }
+
+        [Test]
+        public void ShouldReturnResult()
+        {
+            //Delivery fields
+            this.result.Count().Should().Be(4);
+            var updateData = this.updatedDeliveries.First();
+            var updated = this.result.First();
+            updated.DeliverySeq.Should().Be(1);
```

Comment:
Most of this stuff should already be covered in existing tests so you don't need to test it all again. I would just test that the details quantity is the correct value (along with the monetary totals that depend on the quantity